Automattic/wp-calypso
# State: No longer request unused options in /me/sites

In #47623 we removed the last usage of the `permalink_structure`, `default_sharing_status`, `default_likes_enabled`, `signup_is_store` and `site_goals` options. This PR removes that option from `SITE_REQUEST_OPTIONS` so we won't unnecessarily request that option when we do a `/me/sites`. 

We're not expecting this will have any performance improvement on the `/me/sites` requests, as we're pulling a bunch of other items from the `{$wpdb->prefix}options` tables, but it's generally a good practice to not download data that we don't need 😉 

Props to @jsnajdr for spotting this in https://github.com/Automattic/wp-calypso/pull/47623#issuecomment-732027633.

#### Changes proposed in this Pull Request

* State: No longer request unused options in `/me/sites`

#### Testing instructions

* Manual testing not needed - just verify we're not using the following options anywhere anymore in Calypso.
  * `permalink_structure`
  * `default_likes_enabled`
  * `default_sharing_status`
  * `signup_is_store`
  * `site_goals`
* Verify all tests pass.
